### DISCUSSION TITLE
Lesson 11: Transaction Verification-- Error with cast calldata-decode

### DISCUSSION THREAD

**Discussion by mrthedude** [Cyfrin/foundry-full-course-cu#823] [Q&A] [answered]
[Timestamp](https://youtu.be/sas02qSFZ74?t=37727) for the section of Lesson 11 I am referencing

I tried to use ```cast calldata-decode``` on the ```deposit``` function of the testnet WETH contract Patrick was using as an example ([token address link](https://sepolia.etherscan.io/token/0xdd13e55209fd76afe204dbda4007c227904f0a81#writeContract)) to test out decoding on a function with no parameters. 

After initializing a transaction to deposit 1 testnet WETH,  the calldata that populates in my transaction window is only 4 bytes long (the function selector and signature are the same string value). I used ```cast sig``` to confirm the function signature but when I use ```cast calldata-decode``` and input the signature and calldata, I get the error ```no data was decoded```

I checked the [Function Signature Database](https://openchain.xyz/signatures?query=0xd0e30db0) too just to see if there are any duplicate signatures but it's the only function listed.

Not a road block to completing the course but it has me curious on why I can't decode the calldata. 

Thanks in advance for taking the time to read this and help out my learning! 🙏 🙏 

![Screenshot 2023-09-07 155123](https://github.com/Cyfrin/foundry-full-course-f23/assets/95384358/5e1cca35-e4d3-4c76-9d17-aca70664ba0e)

**Comment by Niferu** [comment, marked as answer]
Hello @mrthedude

You should be using `calldata-decode` only for functions that takes arguments. Look below example:
`cast --calldata-decode "transferFrom(address,address,uint256)" 0x6f760f410000000000000000000000000000000000000000000000000000000000000040000000000000000000000000000000000000000000000000000000000000029a000000000000000000000000000000000000000000000000000000000000000c4c616d612044656c205265790000000000000000000000000000000000000000`

If you want to decode `deposit()` to check if it is really deposit function just use below:
`cast calldata "deposit()"`

**Reply by mrthedude** [reply]
That makes sense, thankyou for the clarification!
